### PR TITLE
[#65] Revive react-transform

### PR DESCRIPTION
**Pull request by nkbt**
react-transform updated to version 2 (still beta) and supports Babel 6, so we can put it back to babel config. Alongside with redbox and hmr.

### CODE REVIEW

**Comment by ericelliott** [comment]
Is the node engine > 5 necessary?

**Comment by nkbt** [comment]
Not required, I can do `>=4`

File: package.json [hunk 1]
```diff
@@ -33,22 +40,25 @@
...
-    "babel-core": "6.3.17",
-    "babel-eslint": "4.1.6",
+    "babel-core": "6.3.21",
+    "babel-eslint": "5.0.0-beta6",
     "babel-loader": "6.2.0",
-    "babel-plugin-react-transform": "1.1.1",
+    "babel-plugin-react-transform": "2.0.0-beta1",
     "babel-plugin-transform-runtime": "6.3.13",
-    "babel-preset-es2015": "6.3.13",
+    "babel-preset-es2015": "^6.3.13",
```
Comment (nkbt) [inline package.json]: will remove `^` too

**Comment by ericelliott** [comment]
Cool. Ping when it's done. Happy to merge. =)

**Comment by nkbt** [comment]
Relaxed Node to >=4, rebased and updated deps again.

**Pull request merged**